### PR TITLE
ft-user-signup:user should be able to sign up

### PR DESCRIPTION
**Pull request by athanase-matabaro**
# PR Details
user should be able to sign up to the blog

## Description
first endpoint:
- create the routing system
- implement joi as middleware to validate the request
- create helper response to configure server responses
- create helper password to hash the password using bcrypt
- create helper token to create a token at authentication layer using jwt
- create controller for the authentication part
- create services for the database 
- create models and migrations for the authentication part 

## Motivation and Context
this changes are required for the user to be able sign up in the blog 

## How Has This Been Tested
using postman for sending request and laragon for the database environment  

## Types of changes

- [x ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ x] All new and existing tests passed.

### CODE REVIEW

File: src/app/middleware/verifysignup.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import UserServices from '../../database/acid/services/userServices';
+import GenericHandler from '../helpers/responses';
+
+const checkUsername = async (req, res, next) => {
+  const { username } = req.body;
+  const usernameExist = await UserServices.findUsername({ username });
+  if (usernameExist) {
+    return GenericHandler.error(400, 'Username already in use!', res);
```
Comment (chrismatgit) [inline src/app/middleware/verifysignup.js:8]: I can suggest you to create a file where you store your status code and their messages. then import them to make clean your code where you want to use them. E.g : `export const HTTP_CREATED_MESSAGE = 'User registered successfully';` and in the other file you can create different status code: `export const HTTP_OK = 200;`. But the rest looks good. Nice work

**Comment by athanase-matabaro** [comment]
thanks but is it mergeable or not ?

**Comment by chrismatgit** [comment]
Sure you can merge.

**Comment by athanase-matabaro** [comment]
okay thanks

**Pull request merged**